tests/php/test-class-dev-tools-user-access.php

```diff
@@ -54,21 +100,71 @@ public function test_register_rest_field() {
 		$this->assertArrayHasKey( 'amp_dev_tools_enabled', $wp_rest_additional_fields['user'] );
 	}
 
+	/**
+	 * Tests DevToolsUserAccess::print_personal_options
+	 *
+	 * @covers DevToolsUserAccess::print_personal_options
+	 */
+	public function test_print_personal_options() {
+		$admin_user  = $this->factory()->user->create_and_get( [ 'role' => 'administrator' ] );
+		$editor_user = $this->factory()->user->create_and_get( [ 'role' => 'editor' ] );
+
+		ob_start();
+		$this->dev_tools_user_access->print_personal_options( $editor_user );
+		$this->assertEmpty( ob_get_clean() );
+
+		wp_set_current_user( $admin_user->ID );
+		ob_start();
+		$this->dev_tools_user_access->print_personal_options( $editor_user );
+		$this->assertEmpty( ob_get_clean() );
+
+		ob_start();
+		$this->dev_tools_user_access->print_personal_options( $admin_user );
+		$this->assertContains( 'checkbox', ob_get_clean() );
+	}
+
+	/**
+	 * Tests DevToolsUserAccess::update_user_setting
+	 *
+	 * @covers DevToolsUserAccess::update_user_setting
+	 */
+	public function test_update_user_setting() {
+		$admin_user  = $this->factory()->user->create_and_get( [ 'role' => 'administrator' ] );
+		$editor_user = $this->factory()->user->create_and_get( [ 'role' => 'editor' ] );
+
+		$_POST[ DevToolsUserAccess::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] = 'true';
+
+		$this->assertFalse( $this->dev_tools_user_access->update_user_setting( $admin_user->ID ) );
+
+		wp_set_current_user( $admin_user->ID );
+		$this->assertFalse( $this->dev_tools_user_access->update_user_setting( $editor_user->ID ) );
+
+		$this->assertTrue( $this->dev_tools_user_access->update_user_setting( $admin_user->ID ) );
+		$this->assertTrue( $this->dev_tools_user_access->get_user_enabled( $admin_user ) );
+		$_POST[ DevToolsUserAccess::USER_FIELD_DEVELOPER_TOOLS_ENABLED ] = null;
+		$this->assertTrue( $this->dev_tools_user_access->update_user_setting( $admin_user->ID ) );
+		$this->assertFalse( $this->dev_tools_user_access->get_user_enabled( $admin_user ) );
+	}
+
 	/**
 	 * Tests DevToolsUserAccess::rest_get_dev_tools_enabled
 	 *
 	 * @covers DevToolsUserAccess::rest_get_dev_tools_enabled
 	 */
 	public function test_rest_get_dev_tools_enabled() {
-		$user = self::factory()->user->create();
+		/** @var WP_User $user */
```

Comment:
`WP_User` is already inferred from the annotation for the method `create_and_get`.

```suggestion
```